projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-custom-dialog.component.ts

```diff
@@ -130,6 +130,7 @@ export class IgxExcelStyleCustomDialogComponent implements AfterViewInit {
                 this.grid.rootGrid ? this.grid.rootGrid.nativeElement : this.grid.nativeElement :
                 esf;
         this.toggle.open(this._customDialogOverlaySettings);
+        this.overlayComponentId = this.toggle.overlayId;
```

Comment:
we should also clear the `overlayComponentId` on close. Otherwise first time the target will be the `esf` and each next time the target will be the grid's ` nativeElement`.

Reply:
Sure, I added a clear. The sample did indeed started opening the dialog with the grid as parent . Still unsure why it works like that but with the clear, the sample seems to be working like before.